src/libraries/Microsoft.PowerFx.Core/IR/BinaryOpMatrix.cs

```diff
@@ -271,9 +271,13 @@ private static BinaryOpKind GetBooleanBinaryOp(PowerFx.Syntax.BinaryOpNode node,
                 case DKind.ObjNull:
                     switch (node.Op)
                     {
-                        case BinaryOp.NotEqual:
+                        case BinaryOp.NotEqual:
+                        case BinaryOp.Less:
+                        case BinaryOp.Greater:
```

Comment:
It feels wrong to use NeqNull for `<` and `>` (and EqNull for `>=` and `<=`). Should we introduce new op kinds?